backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -929,23 +945,204 @@ func addDisplayNameColumn(db *gorm.DB, mdl interface{}, dialect SQLDialect) erro
+	config          *objectstore.Config
+	accessKey       string
+	secretKey       string
+	useDirectBucket bool // indicates if direct blob.OpenBucket should be used (MinIO with env credentials)
+}
+
+// ensureProtocol adds http:// or https:// protocol if not present
+func ensureProtocol(endpoint string, secure bool) string {
+	if strings.HasPrefix(endpoint, "http://") || strings.HasPrefix(endpoint, "https://") {
+		return endpoint
+	}
+	protocol := "http://"
+	if secure {
+		protocol = "https://"
+	}
+	return protocol + endpoint
+}
+
+// buildConfigFromEnvVars creates objectstore.Config from environment variables
+// This bridges the gap between API server's environment variable configuration
+// and the v2 objectstore package's expected configuration format.
+// Returns both the config and credentials to avoid re-reading them.
+func buildConfigFromEnvVars() *blobStorageConfig {
+	bucketName := common.GetStringConfigWithDefault("ObjectStoreConfig.BucketName", "")
+	host := common.GetStringConfigWithDefault("ObjectStoreConfig.Host", "")
+	port := common.GetStringConfigWithDefault("ObjectStoreConfig.Port", "")
+	secure := common.GetBoolConfigWithDefault("ObjectStoreConfig.Secure", false)
+	region := common.GetStringConfigWithDefault("ObjectStoreConfig.Region", "")
+	accessKey := common.GetStringConfigWithDefault("ObjectStoreConfig.AccessKey", "")
+	secretKey := common.GetStringConfigWithDefault("ObjectStoreConfig.SecretAccessKey", "")
+
+	// Constants for MinIO secret (consistent with v2 config)
+	const minioArtifactSecretName = "mlpipeline-minio-artifact"
+	const minioArtifactAccessKeyKey = "accesskey"
+	const minioArtifactSecretKeyKey = "secretkey"
+
+	// Set AWS environment variables
+	if accessKey != "" {
+		os.Setenv("AWS_ACCESS_KEY_ID", accessKey)
+	}
+	if secretKey != "" {
+		os.Setenv("AWS_SECRET_ACCESS_KEY", secretKey)
+	}
+	// Set region - use default for MinIO if not specified
+	isMinIO := host != ""
+	if region != "" {
+		os.Setenv("AWS_REGION", region)
+	} else if isMinIO {
+		// MinIO implements AWS S3's API and uses "us-east-1" as its default region.
+		// This aligns with AWS S3's standard practice and ensures compatibility with
+		// S3 SDKs and clients. Region is required by the AWS SDK but MinIO treats
+		// it as a logical partition rather than a geographic location.
+		os.Setenv("AWS_REGION", "us-east-1")
+	}
+
+	secretNamespace := common.GetPodNamespace()
+
+	endpoint := host
+	if port != "" {
+		endpoint = fmt.Sprintf("%s:%s", host, port)
+	}
+
+	// Build SessionInfo that v2/objectstore.OpenBucket expects
+	// The v2 package will use these parameters to configure the S3 client
+	params := map[string]string{}
+
+	// Configure credential handling based on endpoint type.
+	// MinIO requires special handling as fromEnv=true causes fallback to AWS endpoints.
+	queryString := ""
+	hasCredentials := accessKey != "" && secretKey != ""
+	useDirectBucket := false
+
+	// Set base parameters
+	if isMinIO {
```

Comment:
only this case is allowed, let's get rid of this if statement, and if we are missing these configuration options from, throw an error 

so `useDirectBucket` is always true, and we can remove it